src/vs/editor/common/viewModel/viewModelImpl.ts

```diff
@@ -257,6 +257,11 @@ export class ViewModel extends Disposable implements IViewModel {
 			eventsCollector.emitViewEvent(new viewEvents.ViewDecorationsChangedEvent(null));
 		}
 
+		if (e.hasChanged(EditorOption.renderValidationDecorations)) {
```

Comment:
cc @bpasero fix that @rebornix and I discovered regarding decorations not being updated.